plugins/base/src/main/kotlin/translators/documentables/DescriptionSections.kt

```diff
@@ -0,0 +1,320 @@
+    documentable: Documentable,
+    logger: DokkaLogger,
+) {
+    val inheritors = if (documentable is WithScope) documentable.inheritors() else return
+    if (inheritors.values.none()) return
+
+    // split content section for the case:
+    // parent is in the shared source set (without expect-actual) and inheritor is in the platform code
+    if (documentable.isDefinedInSharedSourceSetOnly(inheritors.keys.toSet()))
+        sharedSourceSetOnlyInheritorsSectionContent(inheritors, logger)
+    else
+        multiplatformInheritorsSectionContent(documentable, inheritors, logger)
+}
+
+/**
+ * Detect that documentable is located only in the shared code without expect-actuals
+ * Value of `analysisPlatform` will be [Platform.common] in cases if a source set shared between 2 different platforms.
+ * But if it shared between 2 same platforms (e.g. jvm("awt") and jvm("android"))
+ * then the source set will be still marked as jvm platform.
+ *
+ * So, we also try to check if any of inheritors source sets depends on current documentable source set.
+ * that will mean that the source set is shared.
+ */
+private fun Documentable.isDefinedInSharedSourceSetOnly(inheritorsSourceSets: Set<DokkaConfiguration.DokkaSourceSet>) =
+    sourceSets.size == 1 &&
+            (sourceSets.first().analysisPlatform == Platform.common
+                    || sourceSets.first().hasDependentSourceSetInInheritors(inheritorsSourceSets))
+
+private fun DokkaConfiguration.DokkaSourceSet.hasDependentSourceSetInInheritors(
+    inheritorsSourceSets: Set<DokkaConfiguration.DokkaSourceSet>,
+) =
+    inheritorsSourceSets.any { sourceSet -> sourceSet.dependentSourceSets.any { it == this.sourceSetID } }
+
+private fun PageContentBuilder.DocumentableContentBuilder.multiplatformInheritorsSectionContent(
+    documentable: Documentable,
+    inheritors: Map<DokkaConfiguration.DokkaSourceSet, List<DRI>>,
+    logger: DokkaLogger,
+) {
+    // intersect is used for removing duplication in case of merged classlikes from different platforms
+    val availableSourceSets = inheritors.keys.toSet().intersect(documentable.sourceSets)
+
+    header(KDOC_TAG_HEADER_LEVEL, "Inheritors", sourceSets = availableSourceSets)
+    table(
+        kind = ContentKind.Inheritors,
+        sourceSets = availableSourceSets,
+        extra = mainExtra + SimpleAttr.header("Inheritors")
+    ) {
+        availableSourceSets.forEach { sourceSet ->
+            inheritors[sourceSet]?.forEach { classlike: DRI ->
+                inheritorRow(classlike, logger, sourceSet)
+            }
+        }
+    }
+}
+
+// `sourceSets` parameters is not applied on purpose
+private fun PageContentBuilder.DocumentableContentBuilder.sharedSourceSetOnlyInheritorsSectionContent(
+    inheritors: Map<DokkaConfiguration.DokkaSourceSet, List<DRI>>,
+    logger: DokkaLogger,
+) {
+    val uniqueInheritors = inheritors.values.flatten().toSet()
+
+    header(KDOC_TAG_HEADER_LEVEL, "Inheritors")
```

Comment:
I suppose you can use `PageContentBuilder.DocumentableContentBuilder.block` function or your own one here and in another functions instead of the duplication:
```
header(KDOC_TAG_HEADER_LEVEL, "Inheritors")
    table(
        kind = ContentKind.Inheritors,
        extra = mainExtra + SimpleAttr.header("Inheritors")
    ) {
        /// some operation
    }
}
```